src/libsyntax/tokenstream.rs

```diff
@@ -820,98 +929,108 @@ impl Index<usize> for TokenStream {
     type Output = TokenTree;
 
     fn index(&self, index: usize) -> &TokenTree {
```

Comment:
I worry a bit about exposing this API. Maybe we should keep it for internal use, in case we end up using a non-uniform encoding? It's not a big deal for now, though, just potential for misuse.